File: examples/PythonExample/print_cb2xml.py
#!/usr/bin/python
##############################################################
#
#  Purpose: Illustrate processing a cb2xml xml file in python.
#           It prints the Xml content to the terminal
#   Author: Bruce Martin
#  License: Whatever: LGPL (any), Apache, Creative commons, etc
#           Use at your own risk
#
##############################################################
import xml.etree.ElementTree as ET

## Get an attribute if it exist otherwise return ''
def getAttr(d, key):
    r=''
    if key in d.attrib:
        r = d.attrib[key]
    return r

# Get attribute if it exists
def getAttrId(d, key):
    r=''
    if key in d.attrib:
        r = key + '=' + d.attrib[key]
    return r
 
##########################################################################
# Purpose: Print one item
##########################################################################
def printItem(indent, item):
    n = indent + item.attrib['level'] + " " + item.attrib['name'] + "                                                                                    "
    n = n[:50]

    print n, '\t', item.attrib['position'], '\t', item.attrib['storage-length'], '\t', getAttr(item, 'display-length'), getAttr(item, 'picture'), getAttrId(item, 'usage'), getAttrId(item, 'numeric'), getAttrId(item, 'signed')
    for child in item.findall('item'):
	printItem(indent + "    ", child)

#########################################################################

tree = ET.parse('cbl2xml_Test110.cbl.xml')
root = tree.getroot()

print ">> ", root.tag, root.attrib['filename']

for child in root.findall('item'):
    printItem("  ", child)
    

File: examples/RubyExample/print_cb2xml.rb
#!/usr/bin/ruby
require 'nokogiri'

################################################################
## This is a simple example of accessing cb2xml from ruby
## It simply prints the content of the Xml
## Author: Bruce Martin
## License: Any, e.g. LGPL (any), Apache, Creative Commons
################################################################
   
def printXml indent, tree

    tree.each do |node|
    	 ##next if !node["position"].nil?

    	 puts "#{indent} #{node["level"]} #{node["name"]}\t#{node["position"]}\t#{node["storage-length"]}\t#{node["picture"]}\t#{node["usage"]}"   
    	 printXml "#{indent}   ", node.children   
    end
end

   doc = Nokogiri.XML(File.open("cb2xml_Output110.xml"))
   
   printXml "    ", doc.root.children
